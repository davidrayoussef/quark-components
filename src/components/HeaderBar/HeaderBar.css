.header {
  display: flex;
  flex: 0 0 auto;
  width: 100%;
  height: 57px;
  padding: 15px 25px;
  color: #EEE;
  background: var(--primary);
  z-index: 1;
}

.header h1 {
  font-weight: 400;
  font-size: 1.2rem;
  margin: 0;
}
