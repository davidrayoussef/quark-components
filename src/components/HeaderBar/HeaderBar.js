import React from 'react';
import PropTypes from 'prop-types';
import style from './HeaderBar.css';

const HeaderBar = ({ title }) => {
  return (
    <header className={style.header}>{title}</header>
  );
};

HeaderBar.proptypes = {
  title: PropTypes.string
};

export default HeaderBar;
