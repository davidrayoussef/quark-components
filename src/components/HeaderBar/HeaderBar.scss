.header {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 57px;
  padding: 15px;
  color: #eee;
  background: var(--primary);
  z-index: 1;
}

.header h1 {
  font-weight: 400;
  font-size: 1.2rem;
  margin: 0;
}

.header a {
  color: unset;
  text-decoration: none;
}

.nav {
  display: flex;
  align-items: center;
  gap: 1rem;
  margin: 0 auto;
  & a {
    transition: all 0.15s;
    &:hover {
      background: none;
      color: var(--secondary);
    }
  }
}
