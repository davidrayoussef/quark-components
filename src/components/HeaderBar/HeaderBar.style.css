header {
  display: flex;
  flex: 0 0 auto;
  padding: 15px;
  margin-bottom: 20px;
  color: #EEE;
  background: #282938;
}
