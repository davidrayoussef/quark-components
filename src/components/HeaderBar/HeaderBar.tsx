import React from 'react';

import { HeaderBarProps } from '@/shared';

import style from './HeaderBar.scss';

export const HeaderBar: React.FC<HeaderBarProps> = ({
  children,
  title,
  navLinks
}: HeaderBarProps) => {
  return (
    <header className={style.header}>
      {title && <h1>{title}</h1>}
      {navLinks && <nav className={style.nav}>{navLinks}</nav>}
      {children}
    </header>
  );
};
