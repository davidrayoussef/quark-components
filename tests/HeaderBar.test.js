import React from 'react';
import { shallow } from 'enzyme';
import HeaderBar from '../src/components/HeaderBar/HeaderBar';

test('should render', () => {
  const wrapper = shallow(<HeaderBar />);
  expect(wrapper);
});

test('should render a title', () => {
  const wrapper = shallow(
    <HeaderBar title="HeaderBar Title" />
  );
  expect(wrapper.text()).toBe('HeaderBar Title');
});

test('snapshot', () => {
  const wrapper = shallow(
    <HeaderBar title="HeaderBar Title" />
  );
  expect(wrapper).toMatchSnapshot();
});
