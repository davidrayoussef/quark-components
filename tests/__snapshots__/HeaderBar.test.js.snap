// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  className="header"
>
  <h1>
    HeaderBar Title
  </h1>
</header>
`;
